pkg/auth/providers/azure/azure_provider_test.go

```diff
@@ -225,3 +228,73 @@ func TestTransformToAuthProvider(t *testing.T) {
 		})
 	}
 }
+
+func TestProcessConfigOnTestAndApply(t *testing.T) {
```

Comment:
So this does test your new function - however, test and apply (which is large portion of logic), remains untested. Given that this is a public facing api endpoint, that's probably not what we want long-term. I realize this may be tough, so I'm willing to let it pass if we need to, but I wanted to make sure that it was brought up.